selftests/pre_release/jobs/features.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import time
+
+from avocado import Test
+from avocado.core import exit_codes
+from avocado.core.job import Job
+
+BOOLEAN_ENABLED = 'on'
+BOOLEAN_DISABLED = 'off'
+
+
+class JobAPIFeaturesTest(Test):
+
+    def check_exit_code(self, exit_code):
+        """Check if job ended with success"""
+        expected_exit_code = self.params.get('exit_code',
+                                             default=exit_codes.AVOCADO_ALL_OK)
+        self.assertEqual(expected_exit_code, exit_code)
+
+    def check_file_exists(self, file_path, assert_func):
+        """Check if a file exist or not depending on the `assert_func`"""
+        assert_func(os.path.exists(file_path))
+
+    def check_file_content(self, file_path, assert_func, content):
+        """
+        Check if `content` exist or not in a file depending on the
+        `assert_func`
+        """
+        assert_func(self.file_has_content(file_path, content))
+
+    @staticmethod
+    def file_has_content(file_path, content):
+        """Check if a file has `content`."""
+        if os.path.isfile(file_path):
+            with open(file_path, "r") as f:
+                source_content = f.read()
+            if content in source_content:
+                return True
+        return False
+
+    def get_assert_function(self, value):
+        """Return an assert function depending on the value passed"""
+        if value == BOOLEAN_ENABLED:
+            assert_func = self.assertTrue
+        else:
+            assert_func = self.assertFalse
+        return assert_func
```

Comment:
I thought about using the same as you are proposing in the beginning, but this method will be responsible for retrieving all the used asserts in the test, so this list will increase when more and more features are added. If the list becomes long, I have a personal preference for keeping a single `return` for ease of reading and maintaining.

Anyway, let's see how it will look like when more features are added to the test.